Core/clim-core/panes.lisp

```diff
@@ -363,11 +363,46 @@ order to produce a double-click")
   (print-unreadable-object (pane sink :type t :identity t)
     (prin1 (pane-name pane) sink)))
 
+(defgeneric find-concrete-pane-class (pane-realizer pane-type &optional errorp)
+  (:documentation "Resolves abstract pane name to a concrete pane
+class. Methods should specialize on the first argument.")
```

Comment:
> Resolves abstract pane name

I still think this should say "the abstract pane type PANE-TYPE …" or something similar. Specifically not "pane name".

> Methods should specialize on the first argument.

Why not the second? Maybe a frame manager has a special translation for one abstract pane type. Why shouldn't that be solved with an `eql`-specialized method?

Please document the `errorp` parameter.

Reply:
I've updated a documentation. As of method specializations: because it is easier to navigate the code. Spaghetti-like generic functions are easy to write, not so easy to understand. This is a simplification measure taken by a clostrophobic person.

Reply:
> As of method specializations: because it is easier to navigate the code. Spaghetti-like generic functions are easy to write, not so easy to understand. This is a simplification measure taken by a clostrophobic person.

That doesn't make sense to me. The logical conclusion would be avoiding generic functions alltogether and always using `(defun (…) (typecase …))`.

Reply:
I can see that. In case of this function indeed that should be an ordinary function; latitude for having specialized panes is already granted by `make-pane-1`, the only reason we complicate make-pane protocol even further is because of mechanism you seem to need. I will look again into the other pull request and see, if it can't be achieved another way.

Regarding this particular function, it is the frame manager who find a pane, so this parameter should be specialized to allow extensions by backends. Having too many moving parts for sake of definition freedom is just unnecessary baggage.